akka-docs/rst/java/code/docs/http/javadsl/server/directives/CustomHttpMethodExamplesTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2015-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package docs.http.javadsl.server.directives;
+
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.event.LoggingAdapter;
+import akka.event.NoLogging$;
+import akka.http.javadsl.ConnectHttp;
+import akka.http.javadsl.Http;
+import akka.http.javadsl.ServerBinding;
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.settings.ParserSettings;
+import akka.http.javadsl.settings.ServerSettings;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.http.scaladsl.model.RequestEntityAcceptance;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Flow;
+import org.junit.Test;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+
+import static akka.http.javadsl.model.HttpProtocols.HTTP_1_0;
+
+public class CustomHttpMethodExamplesTest extends JUnitRouteTest {
+
+  @Test
+  public void testComposition() throws InterruptedException, ExecutionException {
+    ActorSystem  system = system();
+    Materializer materializer = materializer();
+    LoggingAdapter loggingAdapter = NoLogging$.MODULE$;
+
+    int    port = 9090;
+    String host = "127.0.0.1";
+
+    //#customHttpMethod
+    HttpMethod BOLT =
+      HttpMethods.custom("BOLT", false, true, RequestEntityAcceptance.Expected$.MODULE$);
```

Comment:
I committed a new version with**OUT** the above Java class, but ...

Does your suggestion mean we should create a new **Java** abstract class, RequestEntityAcceptance? I think what happens with that is below - let me know if I misunderstood anything. 

[HttpMethod.custom()](https://github.com/akka/akka/blob/65a9e0a0f9372d2e2ecb945c8a9dae129a438f77/akka-http-core/src/main/java/akka/http/javadsl/model/HttpMethods.java#L31L32) changes its signature to take the javadsl's RequestEntityAcceptance from scaladsl's RequestEntityAcceptance. However, as it calls ` akka.http.scaladsl.model.HttpMethod.custom` internally, then scaladsl `HttpMethod.custom()` has to accept javadsl's `RequestEntityAcceptance `...?